superset/db_engine_specs.py

```diff
@@ -984,14 +984,22 @@ def convert_to_hive_type(col_type):
             return tableschema_to_hive_types.get(col_type, 'STRING')
 
         table_name = form.name.data
+        schema_name = form.schema.data
+
+        if '.' in table_name and schema_name:
+            raise Exception(
+                "You can't specify a namespace both in the name of the table "
+                'and in the schema field. Please remove one')
         if config.get('UPLOADED_CSV_HIVE_NAMESPACE'):
-            if '.' in table_name:
+            if '.' in table_name or schema_name:
```

Comment:
If we're ok with not enforcing using the required namespace, isn't something like this more compact and clearer?
```
schema_name = form.schema.data or schema_name or config.get('UPLOAD_CSV_HIVE_NAMESPACE')

if schema:
    if '.' in table_name:
        raise Exception("You can't specify both an explicit schema and the schema in the table name")
    
    table_name = '{}.{}'.format(schema, table_name)
```

Reply:
Because there are two ways to specify the namespace, we need to check if either method was used before loading the table.